src/Interpreters/AsynchronousInsertQueue.cpp

```diff
@@ -213,7 +213,7 @@ AsynchronousInsertQueue::AsynchronousInsertQueue(ContextPtr context_, size_t poo
 
     for (size_t i = 0; i < pool_size; ++i)
         queue_shards[i].busy_timeout_ms
-            = std::min(Milliseconds(settings.async_insert_busy_timeout_min_ms), Milliseconds(settings.async_insert_busy_timeout_max_ms));
+            = std::max(Milliseconds(settings.async_insert_busy_timeout_min_ms), Milliseconds(settings.async_insert_busy_timeout_max_ms));
```

Comment:
hmm this is not clear to me
Wouldn't we have the behavior we wanted to avoid with adaptive timeouts, i.e. the initial inserts would wait `async_insert_busy_timeout_max_ms` for no reason.